File: test_data/code/python/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = set('aeiouAEIOU')
        
        half = len(s) // 2
        
        a = set(s[:half]).intersection(vowels)
        b = set(s[half:]).intersection(vowels)
        
        return len(a) == len(b)

File: test_data/code/java/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int mid = s.length() / 2;
        String half1 = s.substring(0, mid);
        String half2 = s.substring(mid);
        int count1 = 0, count2 = 0;

        for (char c : half1.toCharArray()) {
            if (isVowel(c)) {
                count1++;
            }
        }
        for (char c : half2.toCharArray()) {
            if (isVowel(c)) {
                count2++;
            }
        }

        return count1 == count2;
    }

    private boolean isVowel(char c) {
        c = Character.toLowerCase(c);
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
}